src/include/cephfs/libcephfs.h

```diff
@@ -119,19 +119,23 @@ struct ceph_snapdiff_entry_t {
 };
 
 struct ceph_ll_io_info {
-  void (*callback) (struct ceph_ll_io_info *cb_info);
-  void (*release) (void *);
-  void *release_data;
-  void *priv; // private for caller
-  struct Fh *fh;
-  struct iovec *iov;
-  int iovcnt;
-  int iovmax; // maximum iovcnt is allowed, 0 for no limit
-  int64_t off;
-  int64_t result;
-  bool write;
-  bool fsync;
-  bool syncdataonly;
+	void (*callback) (struct ceph_ll_io_info *cb_info);
+	void *priv; // private for caller
+	struct Fh *fh;
+	struct iovec *iov;
+	int iovcnt;
+	int64_t off;
+	int64_t result;
+	bool write;
+	bool fsync;
+	bool syncdataonly;
+};
+
+struct ceph_ll_io_info_v2 {
```

Comment:
I guess this has to be done this way since this is a C-compat header (else, subclassing would have been a neat approach), yes?

Reply:
true, C header implies this way of API versioning